### PR TITLE
add cli property to universe resource model

### PR DESCRIPTION
**Pull request by tamarrow**
WIP for universe changes for specifying binary clis

### CODE REVIEW

**Comment by BenWhitehead** [comment]
I forgot two things.

1. We need to create the `V2Resource` and `V3Resource` objects for each of the respective `V*Package` objects.
2. Please also copy the updated universe schema into `cosmos-model/src/resources/com/mesosphere/universe/v3/repo/meta/schema` so we have it in the code.

**Comment by tamarrow** [comment]
@BenWhitehead updated

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/UniverseClient.scala [hunk 1]
```diff
@@ -117,6 +118,12 @@ class UniverseRepositoryFetcher(adminRouter: AdminRouter)(implicit statsReceiver
         }
     }
   }
+  implicit def v2ResourceToV3Resource(r2: Option[V2Resource]): Option[V3Resource] = {
+    r2 match {
+      case None => None
+      case Some(r) => Some(V3Resource(r.assets, r.images))
+    }
+  }
```
Comment (tamarrow) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/UniverseClient.scala]: this is temporary, @cruhland will fix with his changes.

File: cosmos-model/src/main/scala/com/mesosphere/universe/v3/model/Resource.scala [hunk 2]
```diff
@@ -3,7 +3,13 @@ package com.mesosphere.universe.v3.model
 /**
   * Conforms to: https://github.com/mesosphere/universe/blob/version-3.x/repo/meta/schema/v3-repo-schema.json#/definitions/resource
   */
-case class Resource(
+case class V2Resource(
   assets: Option[Assets] = None,
   images: Option[Images] = None
 )
+
+case class V3Resource(
```
Comment (cruhland) [inline cosmos-model/src/main/scala/com/mesosphere/universe/v3/model/Resource.scala]: The `v3` model doesn't use the `V2Resource`, does it? If not, then these should be declared in different packages:

- `com.mesosphere.universe.v2.model.Resource`
- `com.mesosphere.universe.v3.model.Resource`

Comment (cruhland) [inline cosmos-model/src/main/scala/com/mesosphere/universe/v3/model/Resource.scala]: Ah, I see. `V2Resource` is used in `V2Package`. Never mind!

Comment (cruhland) [inline cosmos-model/src/main/scala/com/mesosphere/universe/v3/model/Resource.scala]: Hmm, actually, based on the other fields in `V2Package`, maybe we can have just one `Resource` type after all. It just means that whenever `Resource` appears in a `V2Package`, its `cli` field will be `None`.

Comment (cruhland) [inline cosmos-model/src/main/scala/com/mesosphere/universe/v3/model/Resource.scala]: Never mind again. Discussed with Tamar, the current changes are correct.

**Comment by cruhland** [comment]
LGTM :+1:

**Pull request merged**